warehouse/accounts/forms.py

```diff
@@ -117,3 +117,40 @@ def validate_password(self, field):
         if userid is not None:
             if not self.user_service.check_password(userid, field.data):
                 raise wtforms.validators.ValidationError("Invalid password.")
+
+
+class RecoverPasswordForm(LoginForm):
+
+    def __init__(self, *args, **kwargs):
+        super(ResetPasswordForm, self).__init__(*args, **kwargs)
+
+        # Instead of again defining username field, we are using LoginForm
+        # to get the username field and poping password field
+        self._fields.pop('password')
+
+
+class ResetPasswordForm(CredentialsMixin, forms.Form):
+
+    password_confirm = wtforms.PasswordField(
+        validators=[
+            wtforms.validators.DataRequired(),
+            wtforms.validators.EqualTo(
+                "password", "Passwords must match."
+            ),
+        ],
+    )
+
+    def __init__(self, *args, user_name, **kwargs):
+        super(ResetPasswordForm, self).__init__(*args, **kwargs)
+
+        self.username = user_name
+        # Instead of again defining username field, we are using LoginForm
+        # to get the username field and poping password field
+        self._fields.pop('username')
+
+    def validate_password(self, field):
+        userid = self.user_service.find_userid(self.username)
+        if userid is not None:
+            if self.user_service.check_password(userid, field.data):
+                raise wtforms.validators.ValidationError(
+                    "Password shouldn't match with previous one.")
```

Comment:
When I forget a password and then click a link to reset it and set the old password, why it is not ok to have the same password as before?